frontend/src/metabase/visualizations/echarts/cartesian/model/series.ts

```diff
@@ -0,0 +1,150 @@
+// This workaround is necessary for generating a compatible key with `keyForSingleSeries` function,
+// ensuring the correct retrieval of series visualization settings based on the provided `seriesVizSettingsKey`.
+// Will be replaced with just a string key when implementing the dynamic line/area/bar.
+const createLegacySeriesObjectKey = (seriesVizSettingsKey: string) => ({
+  card: {
+    _seriesKey: seriesVizSettingsKey,
+  },
+});
+
+export const getBreakoutDistinctValues = (
+  data: DatasetData,
+  breakoutIndex: number,
+) => Array.from(new Set<RowValue>(data.rows.map(row => row[breakoutIndex])));
+
+/**
+ * Generates series models for a given card with a dataset.
+ *
+ * @param {SingleSeries} singleSeries - The single card and dataset.
+ * @param {CartesianChartColumns} columns - The columns model for the card.
+ * @param {boolean} isFirstCard - Indicates whether the card is the first card in the
+ *                                case of combined cards on a dashboard.
+ * @param {RenderingContext} renderingContext - The rendering context.
+ * @returns {SeriesModel[]} The generated series models for the card.
+ */
+export const getCardSeriesModels = (
+  { card, data }: SingleSeries,
+  columns: CartesianChartColumns,
+  isFirstCard: boolean,
+  renderingContext: RenderingContext,
+): SeriesModel[] => {
+  const hasBreakout = "breakout" in columns;
+
+  // Charts without breakout have one series per selected metric column.
+  if (!hasBreakout) {
+    return columns.metrics.map(metric => {
+      const vizSettingsKey = getSeriesVizSettingsKey({
+        cardName: card.name,
+        metricColumn: metric.column,
+        isFirstCard,
+      });
+
+      return {
+        cardId: card.id,
+        column: metric.column,
+        columnIndex: metric.index,
+        dataKey: getDatasetKey(metric.column, card.id),
+        vizSettingsKey,
+        legacySeriesSettingsObjectKey:
+          createLegacySeriesObjectKey(vizSettingsKey),
+      };
+    });
+  }
+
+  // Charts with breakout have one series per a unique breakout value. They can have only one metric in such cases.
+  const { metric, breakout } = columns;
+  const breakoutValues = getBreakoutDistinctValues(data, breakout.index);
+
+  return breakoutValues.map(breakoutValue => {
+    const vizSettingsKey = getSeriesVizSettingsKey({
+      cardName: card.name,
+      isFirstCard,
+      breakoutValue,
+      breakoutColumn: breakout.column,
+      formatValue: renderingContext.formatValue,
+    });
+
+    return {
+      cardId: card.id,
+      column: metric.column,
+      columnIndex: metric.index,
+      vizSettingsKey,
+      legacySeriesSettingsObjectKey:
+        createLegacySeriesObjectKey(vizSettingsKey),
+      dataKey: getDatasetKey(metric.column, card.id, breakoutValue),
+      breakoutColumnIndex: breakout.index,
+      breakoutColumn: breakout.column,
+      breakoutValue,
+    };
+  });
+};
+
+export const getDimensionModel = (
```

Comment:
Nit: since this function is very small maybe we can just assign these values inline in `getCartesianChartModel`